pkg/trimaran/targetloadpacking/targetloadpacking.go

```diff
@@ -0,0 +1,257 @@
+		}, nil
+	}
+	if targetLoadPackingArgs, ok := obj.(*pluginConfig.TargetLoadPackingArgs); ok {
+		return targetLoadPackingArgs, nil
+	}
+	return nil, fmt.Errorf("want args to be of type TargetLoadPackingArgs, got %T", obj)
+}
+
+func (pl *TargetLoadPacking) Score(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
+	if err != nil {
+		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v", nodeName, err))
+	}
+
+	metrics := pl.metrics                                    // copy to maintain snapshot lest updateMetrics() updates the value
+	if _, ok := metrics.Data.NodeMetricsMap[nodeName]; !ok { // This means the node is new (no metrics yet) or metrics are unavailable due to 404 or 500
+		return framework.MinNodeScore, nil // Avoid the node by scoring minimum
+		//TODO(aqadeer): If this happens for a long time, fall back to allocation based packing. This could mean maintaining failure state across cycles if scheduler doesn't provide this state
+	}
+
+	var curPodCPUUsage int64
+	for _, container := range pod.Spec.Containers {
+		curPodCPUUsage += PredictUtilisation(&container)
+	}
+	if pod.Spec.Overhead != nil {
+		curPodCPUUsage += pod.Spec.Overhead.Cpu().MilliValue()
+	}
+
+	var nodeCPUUtilPercent float64
+	var cpuMetricFound bool
+	for _, metric := range metrics.Data.NodeMetricsMap[nodeName].Metrics {
+		if metric.Type == watcher.CPU {
+			nodeCPUUtilPercent = metric.Value
+			cpuMetricFound = true
+		}
+	}
+
+	if !cpuMetricFound {
+		klog.Errorf("cpu metric not found for node %v in node metrics %v", nodeName, metrics.Data.NodeMetricsMap[nodeName].Metrics)
+		return 0, nil
+	}
+	nodeCPUCapMillis := float64(nodeInfo.Node().Status.Capacity.Cpu().MilliValue())
+	nodeCPUUtilMillis := (nodeCPUUtilPercent / 100) * nodeCPUCapMillis
+
+	var missingCPUUtilMillis int64 = 0
+	pl.eventHandler.RLock()
+	defer pl.eventHandler.RUnlock()
+	if _, ok := pl.eventHandler.ScheduledPodsCache[nodeName]; ok {
+		for _, info := range pl.eventHandler.ScheduledPodsCache[nodeName] {
+			// If the time stamp of the scheduled pod is outside fetched metrics window, or it is within metrics reporting interval seconds, we predict util.
+			// Note that the second condition doesn't guarantee metrics for that pod are not reported yet as the 0 <= t <= 2*metricsAgentReportingIntervalSeconds
+			// t = metricsAgentReportingIntervalSeconds is taken as average case and it doesn't hurt us much if we are
+			// counting metrics twice in case actual t is less than metricsAgentReportingIntervalSeconds
+			if info.Timestamp > metrics.Window.End || info.Timestamp <= metrics.Window.End &&
+				(metrics.Window.End-info.Timestamp) < metricsAgentReportingIntervalSeconds {
+				for _, container := range info.Pod.Spec.Containers {
+					missingCPUUtilMillis += PredictUtilisation(&container)
+				}
```

Comment:
I have some questions about this place.  
There are two cases：
1. The pods (Short inter-arrival times)  run at the same time. So we can add it all together.
2. The pods (Short inter-arrival times)  run successively. This situation may not be suitable for adding it all together.

WDYT？

Reply:
Can you explain "This situation may not be suitable for adding it all together." more? Why is it not suitable?